### DISCUSSION TITLE
Is there a way to use zustand outside of the react component?

### DISCUSSION THREAD

**Discussion by LinThitHtwe** [pmndrs/zustand#2333] [Q&A] [answered]
I want to use zustand outside of the react component. Is there a way I could achieve this ?

**Comment by dbritto-dev** [comment]
@LinThit27 on a vanilla project or react project? because you can use zustand outside of react component but on nextjs is impossible

**Reply by H0BB5** [reply]
@dbritto-dev I'm looking for the recommended approach for using zustand outside of a react component in a vanilla react project (not on the server like with next.js)

**Reply by dbritto-dev** [reply, marked as answer]
@H0BB5 you can use it directly from the store  -> `mystore.getState`, `mystore.setState`, and `mystore.store`

**Reply by LinThitHtwe** [reply]
@dbritto-dev  Yes, sadly I'm trying to use outside of the component in NextJs
